sqlmesh/core/engine_adapter/base.py

```diff
@@ -2015,6 +2037,24 @@ def _table_or_view_properties_to_expressions(
             for key, value in table_or_view_properties.items()
         ]
 
+    def _build_partitioned_by_exp(
+        self,
+        partitioned_by: t.List[exp.Expression],
+        *,
+        partition_interval_unit: t.Optional[IntervalUnit] = None,
+        columns_to_types: t.Optional[t.Dict[str, exp.DataType]] = None,
+        catalog_name: t.Optional[str] = None,
+        **kwargs: t.Any,
+    ) -> t.Union[exp.PartitionedByProperty, exp.Property]:
+        raise NotImplementedError("This engine does not support partitioning.")
```

Comment:
This changes semantics from ignoring unsupported properties to failing. I don't think this aligns well with ability to run the same project across different engines.

Reply:
Ok will update to return None instead and handle None

Reply:
Returns None now: https://github.com/TobikoData/sqlmesh/pull/3117/commits/37ab52c49a691e28a287f86815b1a04275851f21